test/nsidc-smap/smap_utils.py

```diff
@@ -0,0 +1,88 @@
+"""Collection of functions used for nsidc-smap regression tests."""
+
+from earthdata_hashdiff import (
+    create_geotiff_hash_file,
+    create_nc4_hash_file,
+    geotiff_matches_reference_hash_file,
+    nc4_matches_reference_hash_file,
+    h5_matches_reference_hash_file,
+)
+from pathlib import Path
+
+
+def file_for_variable(directory: Path, glob: str) -> Path:
+    """Return a Path to the matching single file.
+
+    Find a file in a directory by a glob string.
+
+    Raise error if mulitple files  match.
+    """
+    results = list(directory.glob(glob))
+    if len(results) != 1:
+        raise ValueError(f"Incorrect Number of files found ({len(results)}) for {glob}")
+
+    return results[0]
+
+
+def comparison_function_by_extension(ext: str) -> callable:
+    """Returns the correct function to call for the input file type's extension."""
+
+    compare_function_map = {
+        ".tif": geotiff_matches_reference_hash_file,
+        ".nc4": nc4_matches_reference_hash_file,
+        ".h5": h5_matches_reference_hash_file,
+    }
+    return compare_function_map[ext]
```

Comment:
Not for this PR, but this feels like maybe a utility function that could live in `earthdata-hashdiff` itself maybe? What do you think?

Reply:
I mean two of the functions are the same thing. I would have used `matches_reference_hash_file_using_xarray` if it was exported. I'm not sure if this would be useful for most people? but I'm not opposed.

Reply:
My thinking with the public API was that the use of `xarray` felt a bit like an implementation detail, and folks likely care about "hey, I want to hash a netCDF4 file", rather than "hey I want to use xarray". But we could expose both.

Reply:
Oh, that would be nice. One that basically uses a mapping like this and uses it on the spot. I like that idea a lot.